pkg/build/api/v1beta1/conversion.go

```diff
@@ -264,26 +264,41 @@ func init() {
 				ref.Registry = ""
 				out.ImageTag = ref.String()
 			}
+			if (out.To != nil || len(in.DockerImageReference) > 0) && len(out.Tag) == 0 {
+				out.Tag = imageapi.DefaultImageTag
+			}
 			return nil
 		},
 		func(in *BuildOutput, out *newer.BuildOutput, s conversion.Scope) error {
 			if err := s.Convert(&in.To, &out.To, 0); err != nil {
 				return err
 			}
-			out.Tag = in.Tag
 			out.PushSecretName = in.PushSecretName
 			if len(in.DockerImageReference) > 0 {
 				out.DockerImageReference = in.DockerImageReference
+				out.Tag = in.Tag
```

Comment:
at a minimum i don't think we should be setting it to "latest" if it was not provided in the first place.  the you're going to have an Output with a Tag==latest and no To field, which is weird.